backend/capellacollab/projects/capellamodels/modelsources/git/models.py

```diff
@@ -13,62 +13,47 @@
 from capellacollab.core.database import Base
 
 
-class RepositoryGitInnerModel(BaseModel):
-    path: str
-    entrypoint: str
-    revision: str
-
-    class Config:
-        orm_mode = True
+def validate_path(path: t.Optional[str]):
+    if path:
+        sequence_blacklist = ["..", "%"]
 
+        for sequence in sequence_blacklist:
+            if sequence in path:
+                raise ValueError(
+                    "The provide path contains invalid sequences."
+                )
+    return path
 
-class RepositoryGitModel(BaseModel):
-    name: str
-    model: RepositoryGitInnerModel
-
-    class Config:
-        orm_mode = True
 
+class PostGitModel(BaseModel):
+    path: str
+    entrypoint: str
+    revision: str
+    username: str
+    password: str
 
-class PatchRepositoryGitModel(BaseModel):
-    primary: t.Optional[bool]
+    _validate_path = validator("path", allow_reuse=True)(validate_path)
 
 
-class GetRepositoryGitModel(RepositoryGitModel):
-    id: int
+class PatchGitModel(PostGitModel):
     primary: bool
-    username: t.Union[str, None]
-
-
-class GetRevisionsResponseModel(BaseModel):
-    branches: t.List[str]
-    tags: t.List[str]
-
-
-class GitCredentials(BaseModel):
-    username: t.Union[str, None]
-    password: t.Union[str, None]
-
-
-class GetRevisionModel(BaseModel):
-    url: str
-    credentials: GitCredentials
-
-
-class PostGitModel(RepositoryGitModel):
-    credentials: GitCredentials
 
 
-class NewGitSource(BaseModel):
+class ResponseGitModel(BaseModel):
+    id: int
+    name: str
     path: str
     entrypoint: str
     revision: str
+    primary: bool
     username: str
-    password: str
-
+    password: bool
```

Comment:
I think the path is not validated there. Also this could inherit from the PostGitModel, like this:
```python
class ResponseGitModel(PostGitModel):
    id: int
    name: str
    primary: bool
    password: bool

    @validator("password", pre=True)
    def transform_password(cls, passw: t.Union[str, bool]) -> bool:
        if isinstance(passw, bool):
            return passw
        return passw is not None and len(passw) > 0

    class Config:
        orm_mode = True
```

Reply:
Do we need to validate the path for the response model? To me, the response model is only ever used to return data and should not need to do any validations at all (here the validator is simply used to transfom the password. Oddly enough, pydantic has no transfomation decorator and simply refers to using the validator).
And how is the case handled where the `PostGitModel` has `str` as the password type and the `ResponseGitModel` has `bool`. Does the response model simply override the boolean value from the post model?

Reply:
1. Yes you’re right
2. I don’t know, I tried and it works but I’m not sure what it does under the hood.

That makes two good reasons to leave it like it is.